src/main/fc/fc_core.c

```diff
@@ -485,7 +483,9 @@ void processRx(timeUs_t currentTimeUs)
             DISABLE_FLIGHT_MODE(HEADFREE_MODE);
         }
         if (IS_RC_MODE_ACTIVE(BOXHEADADJ)) {
-            headFreeModeHold = DECIDEGREES_TO_DEGREES(attitude.values.yaw); // acquire new heading
```

Comment:
@ledvinap 
"In old code, activating BOXHEADADJ will in effect disable headfree mode until switch is released. "
that is not correct 
you only "acquire new heading" as long as BOXHEADADJ is active!